### PR TITLE
dropDownMenu with headerBottom

### PR DESCRIPTION
**Pull request by Standupic**

### CODE REVIEW

File: src/Verktygsproffsen/Header.stories.fs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+open Fable.React
+open Fable.React.Props
+open PM.StyleGuide.Storybook
+open PM.StyleGuide.Verktygsproffsen.HeaderTop
+open PM.StyleGuide.Verktygsproffsen.HeaderMedium
+open PM.StyleGuide.Verktygsproffsen.HeaderBottom
+open CommonTypes
+
+let Header =
+    FunctionComponent.Of( fun () ->
```
Comment (mvsmal) [inline src/Verktygsproffsen/Header.stories.fs]: The `Header` should also have Props and pass `AccountType` and `PhoneNumber` to the the `HeaderTop`. We can think of smth like ContextProvider, but for now we need to pass the global props from the most top to the bottom.

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+type Category =
+    {
+        SubMenu : bool
+        Url : string
+        Name : string
+    }
+
+type ListItemProps =
+    {
+        categories : Category list
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: The properties of records should have `PascalCase` naming

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                          [
+                            i [ if c. SubMenu then Class "fa fa-bars" ][]
+                            str c.Name
+                          ]
+                     ]
+            ))
+        )
+
+let HeaderBottom =
+    FunctionComponent.Of(fun (props: Category list) ->
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: Props should never be a list, it must be a special type, which can contain list fields.

File: src/Verktygsproffsen/HeaderDropDownMenu.stories.fs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        Children : Category list
+    }
+
+type ListItemProps =
+    {
+        name : string
+        url : string
+        children : ReactElement list
+        amount : int
+    }
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderDropDownMenu.stories.fs]: `PascalCase`

File: src/Verktygsproffsen/HeaderDropDownMenu.stories.fs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        let state = Hooks.useState(false)
+
+        li [
+               OnMouseEnter (fun _ -> state.update(fun s -> true))
+               OnMouseLeave (fun _ -> state.update(fun s -> false))
+            ]
+             [
+               a [ Href props.url ] [ str props.name ]
+               if not props.children.IsEmpty && state.current then ul [ Class "sub-menu"] props.children
+               span [ ] [ if props.amount <= 0 then str "" else ofInt props.amount ]
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderDropDownMenu.stories.fs]: You can generate the whole span conditionally, not only its contents

File: src/Verktygsproffsen/HeaderDropDownMenu.stories.fs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+let rec toCategoriesHtml (categories : Category list) =
+    categories
+    |> List.map (fun c ->
+        let listItemProps = {name = c.Name; url = c.Url; children = []; amount = 0}
+        match c.Children with
+        | [] -> ListItems listItemProps
+        | children ->
+             let childrenLI = toCategoriesHtml(children)
+             ListItems { listItemProps with children = childrenLI ; amount = childrenLI.Length}
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderDropDownMenu.stories.fs]: this should better be written with this formatting:
```fsharp
ListItems { listItemProps with 
              children = childrenLI
              amount = childrenLI.Length }
```

File: src/Verktygsproffsen/HeaderDropDownMenu.stories.fs [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+               if not props.children.IsEmpty && state.current then ul [ Class "sub-menu"] props.children
+               span [ ] [ if props.amount <= 0 then str "" else ofInt props.amount ]
+             ]
+    )
+
+
+let rec toCategoriesHtml (categories : Category list) =
+    categories
+    |> List.map (fun c ->
+        let listItemProps = {name = c.Name; url = c.Url; children = []; amount = 0}
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderDropDownMenu.stories.fs]: Please, put fields on separate lines, don't use `;`

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+type ListItemProps =
+    {
+        categories : Category list
+        parentState: bool
+        toggleState: unit -> unit
+    }
+
+let ListItems  =
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: This name conflicts with the `ListItems` in `DropDownMenu`. Can we rename it to `MenuItems`?

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+                 [
+                    a []
+                      [
+                        i [ Class "fa fa-bars" ][]
+                        str props.Name
+                      ]
+                    if state.current then DropDownMenu props.Categories
+                 ]
+            fragment[][
+                ReactDom.createPortal(fragment[][
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: do you actually need fragment here as you have just one element inside anyways?

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+type OverlayProps =
+    {
+        CloseHandler : unit -> unit
+        IsShow : bool
+    }
+
+let Overlay =
+    FunctionComponent.Of( fun (props : OverlayProps) ->
+        fragment [] [
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: Do you actually need fragment here?

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+        if props.IsShow then
+                div [ Class "overlay"
+                      OnClick (fun _ -> props.CloseHandler())]
+                    [
+                      str "Hello overlay"
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: Looks like you forgot to remove test code

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+                ]
+                 [
+                    a []
+                      [
+                        i [ Class "fa fa-bars" ][]
+                        str props.Name
+                      ]
+                    if state.current then DropDownMenu props.Categories
+                 ]
+            fragment[][
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: Do you need this fragment?

File: src/Verktygsproffsen/HeaderBottom.stories.fs [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+                 ]
+            ]
+        )
+
+let MenuItem =
+    FunctionComponent.Of(fun (props : MenuItemProps) ->
+        li []
+            [
+                a []
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderBottom.stories.fs]: `Href` is missing

File: src/Verktygsproffsen/HeaderDropDownMenu.stories.fs [hunk 14]
```diff
@@ -0,0 +1,102 @@
+module PM.StyleGuide.Verktygsproffsen.DropDownMenu
+
+open Fable.React
+open Fable.React.Props
+open PM.StyleGuide.Storybook
+
+
+
+// add length
```
Comment (mvsmal) [inline src/Verktygsproffsen/HeaderDropDownMenu.stories.fs]: Remove comment.

File: src/sass/Verktygsproffsen/header/_drop_menu.sass [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+            top: 0
+            width: 290px
+            left: 290px
+            li
+                border-right: none
+                border-left: none
+                border-bottom: 1px solid colors.$colorBorder
+                display: block
+                height: 49px
+                display: flex
```
Comment (mvsmal) [inline src/sass/Verktygsproffsen/header/_drop_menu.sass]: You have both `block` and `flex` for `display`

File: src/Verktygsproffsen/HeaderBottom.stories.fs (hunk 10 above)
Comment (Standupic) [reply src/Verktygsproffsen/HeaderBottom.stories.fs]: Through on first look, it seems that does not need but actually every ReactEment must have a wrapped by div or fragment in this case.

File: src/Verktygsproffsen/HeaderBottom.stories.fs (hunk 9 above)
Comment (Standupic) [reply src/Verktygsproffsen/HeaderBottom.stories.fs]: You are right the upper fragment does not need.

**Pull request merged**